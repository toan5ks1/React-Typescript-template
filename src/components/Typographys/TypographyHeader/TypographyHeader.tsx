import TypographyBase, { TypographyBaseProps } from "../TypographyBase";

export interface TypographyHeaderProps extends TypographyBaseProps {}

const TypographyHeader = (props: TypographyHeaderProps) => {
    return <TypographyBase variant="h6" {...props} />;
};

export default TypographyHeader;
