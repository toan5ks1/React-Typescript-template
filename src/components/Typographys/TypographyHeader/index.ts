export { default } from "./TypographyHeader";
export * from "./TypographyHeader";
