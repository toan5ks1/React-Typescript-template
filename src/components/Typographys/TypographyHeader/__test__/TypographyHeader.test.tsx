import TypographyHeader, { TypographyHeaderProps } from "../TypographyHeader";

import { render, RenderResult, screen } from "@testing-library/react";

describe("<TypographyHeader />", () => {
    let wrapper: RenderResult;

    const props: TypographyHeaderProps = {
        children: "Test",
    };

    beforeEach(() => {
        wrapper = render(<TypographyHeader {...props} />);
    });

    it("should exist children", () => {
        expect(screen.getByText("Test")).toBeInTheDocument();
    });

    it("should match snapshot", () => {
        expect(wrapper.container).toMatchSnapshot();
    });
});
