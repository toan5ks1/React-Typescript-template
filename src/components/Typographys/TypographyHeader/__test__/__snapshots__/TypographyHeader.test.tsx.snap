// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TypographyHeader /> should match snapshot 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
}

<div>
  <h6
    class="MuiTypography-root MuiTypography-h6 emotion-0"
  >
    Test
  </h6>
</div>
`;
